### ISSUE TITLE
Add RMSE loss function

### ISSUE THREAD

**Issue by tartakovsky** [pytorch/pytorch#45112]
## 🚀 Feature
I would like `RMSELoss` class to be added to `torch.nn` module, implemented, for example, as:

```
class RMSELoss(_Loss):
    def __init__(self, size_average=None, reduce=None, reduction: str = 'mean', eps=1e-6) -> None:
        super(RMSELoss, self).__init__(size_average, reduce, reduction)
        self.eps = eps

    def forward(self, input: Tensor, target: Tensor) -> Tensor:
        return torch.sqrt(F.mse_loss(input, target, reduction=self.reduction)  + self.eps)
```

The motivation for `eps` is discussed here:
https://discuss.pytorch.org/t/rmse-loss-function/16540/4
https://discuss.pytorch.org/t/rmse-loss-function/16540/6

## Motivation

The motivation for adding RMSE loss is that it's a popular loss function which is often encountered in Kaggle competitions and real-world problems. Every time it's needed, one needs to re-implement it.

## Alternatives
The two usual alternatives are to either carry around the class similar to the proposed implementation. Or, which I suspect done more often, to do the following:
```
criterion = nn.MSELoss()
loss = torch.sqrt(criterion(x, y))
```

The latter causes the non-obvious issue described in the discussions above.

**Comment by ayushpatidar** [comment]
Hey,
Can I work on this issue. if it is not assigned to anyone
.

**Comment by tartakovsky** [comment]
Actually, there is probably no advantage in having RMSE as a loss function. It's more computationally expensive than MSE and I don't see how it can be helpful to compensate for that. What I actually wanted is ready-made RMSE metric, not a loss function. Not sure if having a library of metrics is in score for pytorch, so feel free to close this if the issue is not useful.
